x-pack/plugins/saved_objects_tagging/server/types.ts

```diff
@@ -14,6 +14,37 @@ export interface ITagsRequestHandlerContext {
   assignmentService: IAssignmentService;
 }
 
+/** @public */
+export interface CreateTagClientOptions {
+  client: SavedObjectsClientContract;
+}
+
+/** @public */
+export interface CreateTagAssignmentServiceOptions {
+  client: SavedObjectsClientContract;
+}
+
+/** @public */
+export interface SavedObjectTaggingStart {
+  /**
+   * Creates a TagClient bound to the provided SavedObject client.
+   */
+  createTagClient: (options: CreateTagClientOptions) => ITagsClient;
```

Comment:
I asked myself the same question, but in the end I preferred the future proof approach. My main argument was that if at some point we want to allow the creation of an assignmentService scoped to a 'fake request', e.g for taskManager, it would be easier to add it with an option map (and therefor use the same option map pattern for createTagClient for consistency)

Also as the Start contract has two APIs (createTagClient and createInternalAssignmentService), I wasn't really sure how to name both with this `asScopedToClient` approach (`tagStart.createTagClientScopedToSoClient` sounds terribly long).

But no strong feeling either, I would be fine either way.

Reply:
Might as well leave as-is then, unless anyone else feels strongly one way or the other